File: lib/work/manychoice/chcekscoremanychoice.dart
import 'package:flutter/material.dart';
import 'package:flutter_esclass_2/Data/Data.dart';
import 'package:http/http.dart' as http;
import 'dart:convert';

class Chcekscoremanychoice extends StatefulWidget {
  final Examset exam;
  final String username;

  const Chcekscoremanychoice({
    super.key,
    required this.exam,
    required this.username,
  });

  @override
  State<Chcekscoremanychoice> createState() => _ChcekscoremanychoiceState();
}

class _ChcekscoremanychoiceState extends State<Chcekscoremanychoice> {
  late Future<Chcekscoremanychoicedata> manyChoiceData;

  Future<Chcekscoremanychoicedata> sendDataToPHP(int autoId, String username) async {
    final response = await http.post(
      Uri.parse('https://www.edueliteroom.com/connect/Chcekscore_manychoice.php'),
      body: {
        'autoId': autoId.toString(),
        'username': username,
      },
    );

    if (response.statusCode == 200) {
      final data = json.decode(response.body);
      return Chcekscoremanychoicedata.fromJson(data);
    } else {
      print("Error: ${response.statusCode}");
      print("Response Body: ${response.body}");
      throw Exception('Failed to load data');
    }
  }

  @override
  void initState() {
    super.initState();
    manyChoiceData = sendDataToPHP(widget.exam.autoId, widget.username);
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('ตรวจสอบคำตอบ'),
      ),
      body: FutureBuilder<Chcekscoremanychoicedata>(
        future: manyChoiceData,
        builder: (context, snapshot) {
          if (snapshot.connectionState == ConnectionState.waiting) {
            return const Center(child: CircularProgressIndicator());
          } else if (snapshot.hasError) {
            return Center(child: Text('Error: ${snapshot.error}'));
          } else if (!snapshot.hasData || snapshot.data!.manychoiceData.isEmpty) {
            return const Center(child: Text('No data available'));
          } else {
            final data = snapshot.data!;
            
            // คำนวณคะแนนรวม
            double totalScore = data.submitManychoiceData.fold(
              0,
              (sum, item) => sum + item.submitManychoiceScore,
            );

            return Column(
              children: [
                Expanded(
                  child: ListView.builder(
                    itemCount: data.manychoiceData.length,
                    itemBuilder: (context, index) {
                      final manychoice = data.manychoiceData[index];
                      return Card(
                        margin: const EdgeInsets.symmetric(vertical: 8, horizontal: 8),
                        elevation: 4,
                        child: Padding(
                          padding: const EdgeInsets.all(16.0),
                          child: Column(
                            crossAxisAlignment: CrossAxisAlignment.start,
                            children: [
                              Text(
                                'คำถาม ${index + 1}: ${manychoice.manychoiceQuestion} '
                                '(คะแนนเต็ม ${manychoice.manychoiceQuestionScore} คะแนน)',
                                style: const TextStyle(
                                  fontSize: 16,
                                  fontWeight: FontWeight.bold,
                                ),
                              ),
                              const SizedBox(height: 8),
                              _buildChoiceText('ก: ', manychoice.manychoiceA),
                              _buildChoiceText('ข: ', manychoice.manychoiceB),
                              _buildChoiceText('ค: ', manychoice.manychoiceC),
                              _buildChoiceText('ง: ', manychoice.manychoiceD),
                              if (manychoice.manychoiceE != null) _buildChoiceText('จ: ', manychoice.manychoiceE),
                              if (manychoice.manychoiceF != null) _buildChoiceText('ฉ: ', manychoice.manychoiceF),
                              if (manychoice.manychoiceG != null) _buildChoiceText('ช: ', manychoice.manychoiceG),
                              if (manychoice.manychoiceH != null) _buildChoiceText('ซ: ', manychoice.manychoiceH),
                              const SizedBox(height: 8),
                              Text(
                                'คำตอบที่ถูกต้อง: ${manychoice.manychoiceAnswer.toUpperCase()}',
                                style: const TextStyle(color: Colors.green),
                              ),
                              const SizedBox(height: 8),
                              for (var submit in data.submitManychoiceData)
                                if (submit.questionId == manychoice.manychoiceAuto)
                                  Container(
                                    padding: const EdgeInsets.all(8),
                                    decoration: BoxDecoration(
                                      color: submit.submitManychoiceReply.toUpperCase() ==
                                              manychoice.manychoiceAnswer.toUpperCase()
                                          ? Colors.green
                                          : Colors.red,
                                      borderRadius: BorderRadius.circular(8),
                                    ),
                                    child: Text(
                                      'คำตอบที่เลือก: ${submit.submitManychoiceReply}',
                                      style: const TextStyle(
                                        fontSize: 14,
                                        color: Colors.white,
                                      ),
                                    ),
                                  ),
                            ],
                          ),
                        ),
                      );
                    },
                  ),
                ),
                // แสดงคะแนนรวม
                Container(
                  padding: const EdgeInsets.all(16),
                  color: Colors.blue.shade100,
                  child: Text(
                    'คะแนนรวมที่คุณได้: ${totalScore.toStringAsFixed(2)} '
                    'คะแนน จากคะแนนเต็ม ${widget.exam.fullMark} คะแนน',
                    style: const TextStyle(
                      fontSize: 18,
                      fontWeight: FontWeight.bold,
                      color: Colors.black87,
                    ),
                  ),
                ),
              ],
            );
          }
        },
      ),
    );
  }

  Widget _buildChoiceText(String label, String? choice) {
    return choice != null && choice.isNotEmpty
        ? Text(
            '$label $choice',
            style: const TextStyle(fontSize: 14),
          )
        : const SizedBox();
  }
}

File: lib/API/Chcekscore_manychoice.php
<?php
// รวมไฟล์เชื่อมต่อฐานข้อมูล
include 'connect.php';

// รับค่าจาก POST
$autoId = isset($_POST['autoId']) ? $_POST['autoId'] : '';
$username = isset($_POST['username']) ? $_POST['username'] : '';

// ตรวจสอบค่า autoId และ username
if (empty($autoId) || empty($username)) {
    echo json_encode(['error' => 'autoId or username is missing']);
    exit;
}

try {
    // ขั้นตอนที่ 2: ดึงข้อมูลจากตาราง manychoice
    $sql_manychoice = "
        SELECT 
            manychoice_auto, 
            examsets_id, 
            manychoice_question, 
            manychoice_a, 
            manychoice_b, 
            manychoice_c, 
            manychoice_d, 
            manychoice_e, 
            manychoice_f, 
            manychoice_g, 
            manychoice_h, 
            manychoice_answer, 
            manychoice_question_score 
        FROM manychoice 
        WHERE examsets_id = :autoId";
    $stmt_manychoice = $conn->prepare($sql_manychoice);
    $stmt_manychoice->bindParam(':autoId', $autoId, PDO::PARAM_INT);
    $stmt_manychoice->execute();
    $manychoice_data = $stmt_manychoice->fetchAll(PDO::FETCH_ASSOC);

    // ขั้นตอนที่ 3: ดึงข้อมูลจากตาราง submit_manychoice
    $sql_submit_manychoice = "
        SELECT 
            submit_manychoice_auto, 
            examsets_id, 
            question_id, 
            submit_manychoice_reply, 
            submit_manychoice_score, 
            submit_manychoice_time, 
            users_username 
        FROM submit_manychoice 
        WHERE examsets_id = :autoId 
        AND users_username = :username";
    $stmt_submit_manychoice = $conn->prepare($sql_submit_manychoice);
    $stmt_submit_manychoice->bindParam(':autoId', $autoId, PDO::PARAM_INT);
    $stmt_submit_manychoice->bindParam(':username', $username, PDO::PARAM_STR);
    $stmt_submit_manychoice->execute();
    $submit_manychoice_data = $stmt_submit_manychoice->fetchAll(PDO::FETCH_ASSOC);

    // ส่งข้อมูลกลับไปยัง UI
    $response = [
        'manychoice_data' => $manychoice_data,
        'submit_manychoice_data' => $submit_manychoice_data
    ];

    echo json_encode($response);
} catch (PDOException $e) {
    // ถ้ามีข้อผิดพลาดเกิดขึ้น
    echo json_encode(['error' => $e->getMessage()]);
} finally {
    $conn = null; // ปิดการเชื่อมต่อฐานข้อมูล
}
?>
